### PR TITLE
feat: k8s exp status

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
k8s previously did not update job status because we just throw pods at k8s and let it handle the entire lifecycle. This is a quality of life improvement for job queue that allows us to track whether the job is active or pending.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/resourcemanagers/kubernetes_resource_manager.go [hunk 1]
```diff
@@ -214,6 +217,21 @@ func (k *kubernetesResourceManager) receiveRequestMsg(ctx *actor.Context) error
 	case sproto.ResourcesReleased:
 		k.resourcesReleased(ctx, msg.TaskActor)
 
+	case sproto.UpdatePodStatus:
+		var ref *actor.Ref
+		for addr, id := range k.addrToContainerID {
```
Comment (hamidzr) [inline master/internal/resourcemanagers/kubernetes_resource_manager.go:222]: suggestion: would a reverse map of `addrToContainerID map[*actor.Ref]cproto.ID` work better here?  if it's the only usecase?

File: master/internal/sproto/task_actor.go [hunk 2]
```diff
@@ -45,6 +47,12 @@ type (
 		ContainerStopped *TaskContainerStopped
 	}
 
+	// UpdatePodStatus notifies the resource manager of experiment state change.
```
Comment (hamidzr) [inline master/internal/sproto/task_actor.go:50]: nit: "experiment"? maybe job or task would apply better?

**Review by hamidzr — COMMENTED** [review]
lgtm some small coments

File: master/internal/resourcemanagers/kubernetes_resource_manager.go (hunk 1 above)
Comment (eecsliu) [reply master/internal/resourcemanagers/kubernetes_resource_manager.go:222]: There will be a trade-off between using a map here and using a map during deletion. I only expect each to be used once per experiment, so the runtime will be the same, so I chose this design where the code is a bit more compact. We could use two maps as an index and reverse index. Thoughts?

File: master/internal/sproto/task_actor.go (hunk 2 above)
Comment (eecsliu) [reply master/internal/sproto/task_actor.go:50]: This is only applicable to kubernetes though, would it be accurate to associate it with experiments in general?

File: master/internal/resourcemanagers/kubernetes_resource_manager.go (hunk 1 above)
Comment (hamidzr) [reply master/internal/resourcemanagers/kubernetes_resource_manager.go]: that make sense I didn't consider deletions

File: master/internal/sproto/task_actor.go (hunk 2 above)
Comment (hamidzr) [reply master/internal/sproto/task_actor.go:50]: we chatted about this offline

**Review by hamidzr — APPROVED** [review]
:shipit: 
(pods update comment)

File: master/internal/resourcemanagers/kubernetes_resource_manager.go (hunk 1 above)
Comment (eecsliu) [reply master/internal/resourcemanagers/kubernetes_resource_manager.go]: I'll actually use a map during Update Pod Status and let deletion be a linear time operation. In the future, we may add finer grained statuses on k8s, in which case the pod status could be updated multiple times per experiment.

**Pull request merged**